src/jit/optcse.cpp

```diff
@@ -78,18 +78,22 @@ inline Compiler::CSEdsc* Compiler::optCSEfindDsc(unsigned index)
     return optCSEtab[index - 1];
 }
 
-/*****************************************************************************
- *
- *  For a previously marked CSE, decrement the use counts and unmark it
- */
-
-void Compiler::optUnmarkCSE(GenTree* tree)
+//------------------------------------------------------------------------
+// Compiler::optUnmarkCSE
+//    Given a 'tree' return 'true' if the node can safely be removed
+//    or 'false' if the node is a CSE def that the callere needs to
+//    extract and preserve.
+//
+//    If 'tree' is a CSE use then we perform an unmark CSE operation
+//    so that the CSE used counts and weight are updated properly.
+//
+bool Compiler::optUnmarkCSE(GenTree* tree)
```

Comment:
Can you turn this into a full-fledged header comment -- in particular describe what the return value means in the normal way.

Also might be good to mention why this doesn't need to worry about subtrees.